test/metabase/api/notification_test.clj

```diff
@@ -264,3 +269,207 @@
+                                     :recipients   [{:type    :notification-recipient/user
+                                                     :user_id (mt/user->id :lucky)}]}]}]
+      (let [get-notification (fn [user-or-id expected-status]
+                               (mt/user-http-request user-or-id :get expected-status (format "notification/%d" (:id notification))))]
+        (testing "admin can view"
+          (get-notification :crowberto 200))
+
+        (testing "creator can view"
+          (get-notification :rasta 200))
+
+        (testing "recipient can view"
+          (get-notification :lucky 200))
+
+        (testing "other than that no one can view"
+          (get-notification third-user-id 403))))))
+
+(deftest get-system-event-permissions-test
+  (notification.tu/with-system-event-notification!
+    [notification {:event :mb-test/permissions
+                   :notification {:creator_id (mt/user->id :rasta)}}]
+    (let [get-notification (fn [user-or-id expected-status]
+                             (mt/user-http-request user-or-id :get expected-status (format "notification/%d" (:id notification))))]
+      (testing "admin can view"
+        (get-notification :crowberto 200))
+
+      (testing "creator can view"
+        (get-notification :rasta 200))
+
+      (testing "other than that no one can view"
+        (get-notification :lucky 403)))))
+
+(deftest create-card-notification-permissions-test
+  (mt/with-model-cleanup [:model/Notification]
+    (mt/with-user-in-groups [group {:name "test notification perm"}
+                             user  [group]]
+      (mt/with-temp
+        [:model/Card {card-id :id} {:collection_id (t2/select-one-pk :model/Collection :personal_owner_id (mt/user->id :rasta))}]
+        (let [create-notification! (fn [user-or-id expected-status]
+                                     (mt/user-http-request user-or-id :post expected-status "notification"
+                                                           {:payload_type "notification/card"
+                                                            :creator_id   (mt/user->id :rasta)
+                                                            :payload      {:card_id card-id}}))]
+          (mt/with-premium-features #{}
+            (testing "admin can create"
+              (create-notification! :crowberto 200))
+
+            (testing "users who can view the card can create"
+              (create-notification! :rasta 200))
+
+            (testing "normal users can't create"
+              (create-notification! (:id user) 403))
+
+            (mt/when-ee-evailable
+             (testing "users with subscription permissions"
+               (perms/grant-application-permissions! group :subscription)
+
+               (testing "without advanced-permissions enabled"
+                 (testing "cannot create notifications"
+                   (create-notification! (:id user) 403)))
```

Comment:
Even with the premium features enabled, this user can still not create notifications. These test aren't covering the application permissions aspect, and that seems crucial.

Reply:
nice catch, added a test for when an user without subscriptions but can read, will still cannot create a subscription.